Library/Homebrew/cli/named_args.rb

```diff
@@ -103,6 +103,20 @@ def to_formulae_to_casks(only: parent&.only_formula_or_cask, method: nil)
                                                 .map(&:freeze).freeze
       end
 
+      # Returns formulae and casks after validating that a tap is present for each of them.
+      def to_formulae_and_casks_with_taps
+        to_formulae_and_casks.each do |formula_or_cask|
+          case formula_or_cask
+          when Formula
+            odie "Formula #{formula_or_cask.name} is not in a tap!" unless formula_or_cask.tap
+          when Cask::Cask
+            odie "Cask #{formula_or_cask.token} is not in a tap!" unless formula_or_cask.tap
+          else
+            raise ArgumentError, "Expected a formula or a cask: #{formula_or_cask}"
+          end
+        end
+      end
```

Comment:
This message can maybe be improved by adding something about tapping the core taps locally. I'll have to think about it.

Reply:
> is not in any locally tapped taps. You may need to `brew tap` additional taps.

Given this is longer/shared: it pairs nicely with suggestion above.